nsv13/code/modules/overmap/overmap.dm

```diff
@@ -62,61 +74,74 @@
+	var/list/mobs_in_ship = list() //A list of mobs which is inside the ship. This is generated by our areas.dm file as they enter / exit areas
+
+	// Controlling equipment
 	var/obj/machinery/computer/ship/helm //Relay beeping noises when we act
 	var/obj/machinery/computer/ship/tactical
 	var/obj/machinery/computer/ship/dradis/dradis //So that pilots can check the radar easily
+
+	// Ship weapons
 	var/list/railguns = list() //Every railgun present on the ship
 	var/list/torpedo_tubes = list() //every torpedo tube present on the ship.
 	var/list/pdcs = list() //Every PDC ammo rack that we have.
-	var/datum/starsystem/current_system //What starsystem are we currently in? Used for parallax.
-	var/datum/gas_mixture/cabin_air //Cabin air mix used for small ships like fighters (see overmap/fighters/fighters.dm)
-	var/obj/machinery/portable_atmospherics/canister/internal_tank //Internal air tank reference. Used mostly in small ships. If you want to sabotage a fighter, load a plasma tank into its cockpit :)
-	var/resize = 0 //Factor by which we should shrink a ship down. 0 means don't shrink it.
-	var/list/docking_points = list() //Where we can land on this ship. Usually right at the edge of a z-level.
+	var/list/ship_lasers = list() //Every ship-to-ship laser gun we have
+
+	var/fire_mode = FIRE_MODE_PDC //What gun do we want to fire? Defaults to railgun, with PDCs there for flak
 	var/weapon_safety = FALSE //Like a gun safety. Entirely un-used except for fighters to stop brainlets from shooting people on the ship unintentionally :)
-	var/armour_plates = 0 //You lose max integrity when you lose armour plates.
-	var/max_armour_plates = 500 //Placeholder. Set by counting in game objects.
+	var/faction = null //Used for target acquisition by AIs
+
+	var/weapon_range = 10 //Range changes based on what weapon youre using.
+	var/fire_delay = 5
+	var/next_firetime = 0
+
+	var/obj/weapon_overlay/railgun/railgun_overlay
+	var/obj/weapon_overlay/laser/laser_overlay
+	var/atom/last_target //Last thing we shot at, used to point the railgun at an enemy.
+
+	var/torpedoes = 15 //Prevent infinite torp spam
+
+	var/pdc_miss_chance = 20 //In %, how often do PDCs fire inaccurately when aiming at missiles. This is ignored for ships as theyre bigger targets.
+	var/list/torpedoes_to_target = list() //Torpedoes that have been fired explicitly at us, and that the PDCs need to worry about.
 
 /obj/structure/overmap/can_be_pulled(user) // no :)
 	return FALSE
 
-/obj/railgun_overlay //Railgun sits on top of the ship and swivels to face its target
-	name = "Railgun"
-	icon_state = "railgun"
+/obj/weapon_overlay
+	name = "Weapon overlay"
 	layer = 4
 	mouse_opacity = FALSE
 	var/angle = 0 //Debug
 
+/obj/weapon_overlay/railgun //Railgun sits on top of the ship and swivels to face its target
+	name = "Railgun"
+	icon_state = "railgun"
+
+/obj/weapon_overlay/laser
+	name = "Laser cannon"
+	icon = 'icons/obj/hand_of_god_structures.dmi'
+	icon_state = "conduit-red"
```

Comment:
note to self: pixelshift this badboy so it's at the back of the ship or something

this code also means I can have multiple independantly rotating railguns which is gigabased